src/pvcam.cpp

```diff
@@ -550,16 +560,33 @@ PVCamCamera::get_frame(void* im, size_t* nbytes, struct ImageInfo* info)
 
     std::unique_lock<std::mutex> frame_lock(*frame_mutex_);
 
-    // callback will increment frames_available
-    callback_context_.cv.wait_for(
-      frame_lock,
-      std::chrono::microseconds(exposure_time_ * exposure_time_to_us_),
-      [&] { return callback_context_.frames_available > 0; });
+    while (is_running_) {
+        // callback will increment frames_available
+        callback_context_.cv.wait_for(
+          frame_lock,
+          std::chrono::microseconds(exposure_time_ * exposure_time_to_us_),
+          [&] {
+              return callback_context_.frames_available > 0 ||
+                     callback_context_.error;
+          });
+
+        if (callback_context_.error) {
+            // error is logged in the callback handler thread
+            callback_context_.error = false;
+            *nbytes = 0;
+            return;
+        }
+
+        if (callback_context_.frames_available > 0) {
+            break;
+        }
```

Comment:
Nit: couldn't this condition be absorbed into the while's condition?

I.e. use `while (is_running && callback_context_.frames_available == 0)` and remove these three lines?

Reply:
Sure, that's a good idea.